tests/formatter.rs

```diff
@@ -279,9 +279,9 @@ fn test_only_source() {
         .snippet(Snippet::source("").origin("file.rs"));
     let expected = str![[r#"
 error
---> file.rs
- |
- |
+ --> file.rs
+  |
+  |
```

Comment:
I wonder if we shouldn't affect this case.  Maybe we should track `max_lineno` as an `Option` so `Some(0)` can have a width of 1 while `None` can have a width of 0.

Reply:
I will explain my train of thought at #57  to help you estimate it.

Reply:
@epage I have written down my train of thought [here](https://github.com/rust-lang/annotate-snippets-rs/issues/57#issuecomment-2301268251).

In fact, I don't quite understand why `max_lineno` might be `None`. I assume it would definitely be `usize` in my mind.

Reply:
The answer to that question I leave to @Muscraft and @estebank

Reply:
I see. I think that you should check for `self.body.iter().any(|set| set.display_lines.len() > 0)`. If `false`, then you only have a message and file path, so you don't *need* to move right. That case in rustc is rendered:

```
error: message
---> file.rs
```

without any attempt to print the `|`s. It seems to me that there would be left-over work to match that behavior.

That situation can occur when we have a span, but the underlying code couldn't be accessed for printing (this can happen with foreign crates or remapped paths).

All that to say, I think the previous output was better behavior, but I won't block either way.

Reply:
Now I understand what both of you are trying to tell me. It is true that `max_lineno` could be `None`! Thank you.

I've made an update. Please take a look when you have time 🙏.